tools/config.py

```diff
@@ -489,28 +502,90 @@ def regions(self):
         except KeyError:
             raise ConfigException("Not enough information in CMSIS packs to "
                                   "build a bootloader project")
+
+        rom_current_addr = 0
+        FLASH_ERASE_SIZE = self.target.flash_erase_size
+        VECTOR_TABLE_SIZE = self.target.vector_table_size
+
+        # Bootloader and Firmware Metadata Regions
         if 'target.bootloader_img' in target_overrides:
-            filename = target_overrides['target.bootloader_img']
-            part = intelhex_offset(filename, offset=rom_start)
-            if part.minaddr() != rom_start:
+            # find bootloader starting point
+            bootloader_start = rom_start
+
+            # find bootloader size
+            bootloader_fn = target_overrides['target.bootloader_img']
+            bootloader_part = intelhex_offset(bootloader_fn, offset=rom_start)
+            if bootloader_part.minaddr() != rom_start:
                 raise ConfigException("bootloader executable does not "
                                       "start at 0x%x" % rom_start)
-            part_size = (part.maxaddr() - part.minaddr()) + 1
-            yield Region("bootloader", rom_start + start, part_size, False,
-                         filename)
-            start += part_size
+            bootloader_size = (bootloader_part.maxaddr() - bootloader_part.minaddr()) + 1
+
+            # yeild bootloader region
```

Comment:
Please remove redundant comment. This comment just says the same thing that the line of code beneath it says.